### ISSUE TITLE
New parameter - population assignments file

### ISSUE THREAD

**Issue by isaacovercast** [dereneaton/ipyrad#42]
It would be useful on many fronts to have a population assignments file passed in or set as a parameter. Output formats use this (migrate/treemix), D stats uses this, hierarchical clustering would use this, and population summary stats would use this (once i get it going). Standard file format for pop assignment files is normally like this:

          ind1 pop1
          ind2 pop1
          ind3 pop1
          ind4 pop2
          ind5 pop2
          ind6 pop2
          <etc>

Space, or tab separated individuals and pop assignments. Think it's fair game to ask for this?

**Comment by dereneaton** [comment]
Yeah, that seems like a good way to do it. Ideally, the file would be converted to a dictionary. For the API I would like for users to be able to enter the pop assignments as a dictionary directly instead of loading a file. It just seems more Pythonic to not have to write the assignments to a file and then have to read it back in. 

I keep going back and forth in how much we need to allow users to do through the CLI. In the end, they just need to be able to get to the .loci file, and from there if they want to do more complicated things, like population assignments, we could require that they wade into using the API. 

```
## parse populations assignments from a file
data.set_params("pop_assign_path", "popfile.txt")

## or set assignments directly 
data.populations = {'pop1' : [ind1, ind2, ind3], 'pop2' : [ind4, ind5, ind6]}
```

**Comment by isaacovercast** [comment]
We could do it both ways. Setparams could just open the file and dump the dictionary to data.populations, which API users can manipulate directly, and which we will use internally. I think this is a good thing to support for CLI users because it opens up lots of possibilities, it's a familiar concept (from structure, or other progs), and it's pretty low barrier to entry coding wise. I'll work on it.

**Comment by isaacovercast** [comment]
fixed 551cd6f
